### DISCUSSION TITLE
Compile this plugin into Clappr

### DISCUSSION THREAD

**Discussion by ghost** [kslimani/clappr-ima-plugin#37] [Q&A]
hi Guys, how can I integrate (compile) it into clappr plugins? I would like to have it integrated into the player without calling it externally via the script

**Comment by kslimani** [comment]
hi, just import it in your application bundle using npm or another package manager.

**Comment by ghost** [comment]
hi kslimani, I tried to use "npm add clappr-ima-plugin" in the clappr directory. Imported it as a dependency but I don't understand how to call it in plugins to compile it. could you help me figure out how to do it?

**Comment by kslimani** [comment]
Please note this plugin does not support yet the new 0.4.x version of Clappr, it is supported only in the 0.3.x versions (latest is [0.3.13](https://github.com/clappr/clappr/tree/0.3.13))

I am not sure to understand what you want to achieve ? Are you trying to rebuild your own version of Clappr player ?

If it is the case, you need to import the plugin and add it in the [loader class](https://github.com/clappr/clappr/blob/0.3.13/src/components/loader/loader.js#L83), and then recompile the player.

IMO, it is easier to create a wrapper utility class which return your custom player instance. (and should also be easier to maintain).

**Reply by ghost** [reply]
yes I'm trying to compile the player without some plugins and add IMA support with your plugin, but I saw that with version 0.4.4 your plugin works fine using the external script <script src = "...."> .

So would it be enough to add it to the Loader array with the name ClapprImaPlugin and it should be integrated without problems?

I ran the command "npm add clappr-ima-plugin", is this enough to be able to use the plugin in the loader?

**Reply by kslimani** [reply]
While the ima plugin may work on desktop, i cannot guarantee it will work on mobile devices.

Clappr 0.4.x is a complete another story in term of compiling. It has been separated in several projects (and therefore is more complex).

Take a look at main project [dev dependencies](https://github.com/clappr/clappr/blob/dev/package.json#L34) : you could, for example, do the same they did with [hlsjs-playback plugin](https://github.com/clappr/clappr/blob/dev/src/main.js); by importing ima plugin and use loader [registerPlugin](https://github.com/clappr/clappr-core/blob/master/src/components/loader/loader.js#L66) static method. _(do not use registerPlayback, this plugin is not a playback plugin)_.

**Reply by ghost** [reply]
Ok kslimani,
so on this basis I decided to use version 0.3 for mobile compatibility.
In this case just import the plugin with "npm add [package]" and then add it in the loader class or is there some other step to do?

**Reply by kslimani** [reply]
Yep, just import the plugin and add it in the loader like described above, and rebuild the player.
